playbooks/common/openshift-master/config.yml

```diff
@@ -17,6 +17,13 @@
     - node
     - .config_managed
 
+  - fail:
+      msg: "Both openshift_master_session_auth_secrets and openshift_master_session_encryption_secrets must be provided if either variable is set"
+    when: (openshift_master_session_auth_secrets is defined and openshift_master_session_encryption_secrets is not defined) or (openshift_master_session_encryption_secrets is defined and openshift_master_session_auth_secrets is not defined)
+  - fail:
+      msg: "openshift_master_session_auth_secrets and openshift_master_encryption_secrets must be equal length"
+    when: (openshift_master_session_auth_secrets is defined and openshift_master_session_encryption_secrets is defined) and (openshift_master_session_auth_secrets | length != openshift_master_session_encryption_secrets | length)
+
   - set_fact:
       openshift_master_etcd_port: "{{ (etcd_client_port | default('2379')) if (groups.oo_etcd_to_config is defined and groups.oo_etcd_to_config) else none }}"
       openshift_master_etcd_hosts: "{{ hostvars
```

Comment:
As an aside, we use oo_select | oo_collect in quite a few places...  maybe we should add a filter that does both?

something like `hostvars | oo_select_collect(<list of dict keys to select>, <list of attributes to collect, accepting jinja2 'dot' notation>, <list of default values for attributes to collect>)` ?